src/cpp/include/openvino/genai/image_generation/generation_config.hpp

```diff
@@ -82,27 +127,104 @@ struct OPENVINO_GENAI_EXPORTS ImageGenerationConfig {
+ * for better performance as internally compuations will be performed with batch for Unet / Transformer models
+ * and text embeddings tensors will also be computed only once.
+ */
 static constexpr ov::Property<size_t> num_images_per_prompt{"num_images_per_prompt"};
+
+/**
+ * Guidance scale parameter which controls how model sticks to text embeddings generated
+ * by text encoders within a pipeline. Higher value of guidance scale moves image generation towards
+ * text embeddings, but resulting image will be less natural and more augmented.
+ */
 static constexpr ov::Property<float> guidance_scale{"guidance_scale"};
+
+/**
+ * Specifies a height or resulting image. Typically, image height must be divisible by VAE scale factor
+ * (which is 8 in most of case) which represents ratio between latent image / RGB image sizes.
+ */
 static constexpr ov::Property<int64_t> height{"height"};
+
+/**
+ * Specifies a width or resulting image. Typically, image width must be divisible by VAE scale factor
+ * (which is 8 in most of case) which represents ratio between latent image / RGB image sizes.
+ */
 static constexpr ov::Property<int64_t> width{"width"};
+
+/**
+ * Defines a number of inference steps used to denoise initial noised latent to final image.
+ * Note, that in case of image to image / inpainting pipelines, the resulting number of inference steps
+ * is scaled with 'strength' parameter.
+ */
 static constexpr ov::Property<size_t> num_inference_steps{"num_inference_steps"};
 
+/**
+ * Indicates extent to transform the reference `image`. Must be between 0 and 1. `image` is used as a
+ * starting point and more noise is added the higher the `strength`. The number of denoising steps depends
+ * on the amount of noise initially added. When `strength` is 1, added noise is maximum and the denoising
+ * process runs for the full number of iterations specified in `num_inference_steps`. A value of 1
+ * essentially ignores `image`.
+ */
 static constexpr ov::Property<float> strength{"strength"};
 
+/**
+ * Overrides default randon generator used within image generation pipelines.
+ * By default, 'CppStdGenerator' generator is used, but if you are running Image generation via
+ * python code, you can additionally install 'torch' and use OpenVINO GenAI's 'TorchGenerator'
+ * which ensures the generated images will look as in HuggingFace when the same sed value if used.
+ */
 static constexpr ov::Property<std::shared_ptr<Generator>> generator{"generator"};
 
+/**
+ * This parameters limits max sequence length for T5 encoder for SD3 and FLUX models.
+ * Output T5 tokenizer is padded with pad tokens to 'max_sequence_length' within a pipeline.
+ * So, for better performance, you can specify this parameter to lower value to speed-up
+ * T5 encoder inference as well as inference of transformer denoising model.
+ * For optimal performance it can be set to a number of tokens for 'prompt_3' / 'negative_prompt_3' for SD3
+ * or `prompt_2` for FLUX.
```

Comment:
Describe the drawbacks of lowering the value. I guess the pipeline could have computed that value on its own if there were no drawbacks.

Reply:
Seq len 64
![{BBEB462E-6761-4D47-88F6-3D83159DE086}](https://github.com/user-attachments/assets/932acc73-62c2-4a3a-80b7-8f20b31931ab)

Seq len 128
![{222D7E88-4383-4AA2-B718-ADEC8C91591E}](https://github.com/user-attachments/assets/a644a6b5-7d84-495e-afed-0b1b505f36cc)

Seq len 256 
![{1AE6288B-9BB9-4A6A-8ADC-F577A2190E47}](https://github.com/user-attachments/assets/59cbf679-5140-455d-8aa4-49858d714755)

Seq len 512
![{40D6E89D-06AB-4B93-A2F4-9428AC49D196}](https://github.com/user-attachments/assets/50585085-3515-4cb2-b723-111d2434ee34)

I don't know why it's implemented this way in diffusers, but they pad tokenized strings with paddings to be max_sequence_len and then feed to text encoders. We can see from above pictures, that images are close, but still different. 

So, that is why we keep HF aligned behavior, but have this performance tip for users.